File: Math/C/polinomial.c
// Metodo de Horner 
#include <stdio.h>
#include <time.h> 

int horner(int poly[], int n, int x)  { 
    int result = poly[0]; 
 
    for (int i = 1; i < n; i++) 
        result = result * x + poly[i]; 
 
    return result; 
} 

int main() {
    clock_t t; 
    t = clock();

    int poly[] = {2, -6, 2, -1}; 
    int x = 3; 
    int n = sizeof(poly)/sizeof(poly[0]); 

    t = clock() - t; 
    double time_taken = ((double)t)/CLOCKS_PER_SEC;

    printf("Valor do polinômio: %d\n", horner(poly, n, x)); 
    // perceba o quão rápido é
    printf("horner() demorou %f segundos para executar\n", time_taken); 

    return 0; 
}

File: Math/Python/polinomial.py
import time

def horner(poly, x):
    """
    Avalia um polinômio em um determinado valor de x usando o método de Horner.

    Args:
        poly: Uma lista de coeficientes do polinômio, do termo de maior grau ao constante.
        x: O valor para o qual o polinômio será avaliado.

    Returns:
        O valor do polinômio em x.
    """
    result = poly[0]
    for i in range(1, len(poly)):
        result = result * x + poly[i]
    return result

if __name__ == "__main__":
    start_time = time.time()

    poly = [2, -6, 2, -1]
    x = 3

    polynomial_value = horner(poly, x)

    end_time = time.time()
    time_taken = end_time - start_time

    print(f"Valor do polinômio: {polynomial_value}")
    # Perceba o quão rápido é
    print(f"horner() demorou {time_taken:.6f} segundos para executar")
